src/policy/rbf.cpp

```diff
@@ -5,9 +5,10 @@
 #include <policy/rbf.h>
 #include <util/rbf.h>
 
-RBFTransactionState IsRBFOptIn(const CTransaction& tx, const CTxMemPool& pool)
+namespace {
+RBFTransactionState IsRBFOptIn(const CTransaction& tx, const CTxMemPool* const pool) NO_THREAD_SAFETY_ANALYSIS
 {
-    AssertLockHeld(pool.cs);
+    if (pool) AssertLockHeld(pool->cs);
```

Comment:
Could we avoid:
- adding new `NO_THREAD_SAFETY_ANALYSIS`
- conditional `AssertLockHeld()`: https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#no-conditionally-held-locks

?

Reply:
Fixed